Common++/header/IpAddress.h

```diff
@@ -447,6 +463,102 @@ namespace pcpp
 		return *this;
 	}
 
+
+	/**
+	 * @class IPv4Network
+	 * A class representing IPv4 network definition
+	 */
+	class IPv4Network
+	{
+	public:
+		/**
+		 * A constructor that creates an instance of the class out of an address representing the network prefix
+		 * and a prefix length
+		 * @param address An address representing the network prefix. If the address is invalid std::invalid_argument
+		 * exception is thrown
+		 * @param prefixLen A number between 0 and 32 representing the prefix length. If another value is provided
+		 * std::invalid_argument exception is thrown
+		 */
+		IPv4Network(const IPv4Address& address, uint8_t prefixLen);
+
+		/**
+		 * A constructor that creates an instance of the class out of an address representing the network prefix
+		 * and a subnet mask
+		 * @param address An address representing the network prefix. If the address is invalid std::invalid_argument
+		 * exception is thrown
+		 * @param subnetMask A string representing a subnet mask in the format of X.X.X.X, for example: 255.255.0.0.
+		 * Please notice that subnets that start with zeros are invalid, for example: 0.0.255.255. The only subnet mask
+		 * starting with zeros that is valid is 0.0.0.0. If the subnet is invalid std::invalid_argument
+		 * exception is thrown
+		 */
+		IPv4Network(const IPv4Address& address, const std::string& subnetMask);
+
+		/**
+		 * A constructor that creates an instance of the class out of a string representing the network prefix and
+		 * a prefix length or a subnet mask
+		 * @param addressAndSubnet A string in one of these formats:
+		 *  - X.X.X.X/Y where X.X.X.X is a valid IPv4 address representing the network prefix and Y is a number between
+		 *    0 and 32 representing the network prefix
+		 *  - X.X.X.X/Y.Y.Y.Y where X.X.X.X is a valid IPv4 address representing the network prefix and Y.Y.Y.Y is
+		 *    a valid subnet mask
+		 *  For any invalid value std::invalid_argument is thrown
+		 */
+		IPv4Network(const std::string& addressAndSubnet);
+
+		/**
+		 * @return The prefix length, for example: the prefix length of 10.10.10.10/255.0.0.0 is 8
+		 */
+		uint8_t getPrefixLen() const;
+
+		/**
+		 * @return The prefix length, for example: the subnet mask of 10.10.10.10/8 is 255.0.0.0
+		 */
+		std::string getSubnetMask() const { return IPv4Address(m_Mask).toString(); }
+
+		/**
+		 * @return The network prefix, for example: the network prefix of 10.10.10.10/16 is 10.10.0.0
+		*/
+		IPv4Address getNetworkPrefix() const { return IPv4Address(m_NetworkPrefix); }
+
+		/**
+		 * @return The lowest IPv4 address in this network, for example: the lowest address in 10.10.l0.10/16 is
```

Comment:
typo 10.10.l0.10

Reply:
Good catch, thanks!

Reply:
Fixed in https://github.com/seladb/PcapPlusPlus/pull/1103/commits/96ebb3f4dbe61d8511c75a446110ed7da15b153e